azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java

```diff
@@ -95,7 +111,65 @@ public boolean isAvailable() {
 
     public boolean isAuthenticated() {
-        return isAvailable() && this.entity.isAuthenticated();
+        return this.entity.isAuthenticated();
+    }
+
+    protected Mono<Account> authenticateAsync() {
+        return this.verifyAndGetTenantId(createTokenCredential()).map(tenantAndToken -> {
+            if (StringUtils.isBlank(this.entity.getEmail()) && tenantAndToken.getT2() instanceof MsalToken) {
+                String refreshToken = getEmailAndRefreshToken(tenantAndToken.getT2());
+                if (StringUtils.isNotEmpty(refreshToken)) {
+                    AzureEnvironment env = ObjectUtils.firstNonNull(this.entity.getEnvironment(), Azure.az(AzureCloud.class).getOrDefault());
+                    String authority = AzureEnvironmentUtils.getAuthority(env);
+                    this.entity.setTenantCredential(new RefreshTokenTenantCredential(
+                            authority, IdentityConstants.DEVELOPER_SINGLE_SIGN_ON_ID, refreshToken
+                    ));
+                }
+            }
+            if (this.entity.getTenantIds() == null) {
+                this.entity.setTenantIds(tenantAndToken.getT1());
+            }
+            this.entity.setAuthenticated(true);
+            return this;
+        }).flatMap(ignore -> {
+            if (this.entity.getSubscriptions() == null) {
+                return initializeSubscriptions().then(Mono.just(this));
+            }
+            return Mono.just(this);
+        });
+    }
+
+    private String getEmailAndRefreshToken(AccessToken accessToken) {
+        IAuthenticationResult result = ((MsalToken) accessToken).getAuthenticationResult();
+        if (result == null) {
+            return null;
+        }
+
+        if (result.account() != null) {
+            entity.setEmail(result.account().username());
+        }
+
+        String refreshTokenFromResult;
+        try {
+            refreshTokenFromResult = (String) FieldUtils.readField(result, "refreshToken", true);
+        } catch (IllegalAccessException e) {
+            throw new AzureToolkitAuthenticationException("Cannot read refreshToken from IAuthenticationResult.");
+        }
+
+        return refreshTokenFromResult;
+    }
+
+    private Mono<Tuple2<List<String>, AccessToken>> verifyAndGetTenantId(TokenCredential credential) {
+        AzureEnvironment env = ObjectUtils.firstNonNull(this.entity.getEnvironment(), Azure.az(AzureCloud.class).getOrDefault());
+        TokenRequestContext tokenRequestContext = new TokenRequestContext()
+                .addScopes(ScopeUtil.resourceToScopes(env.getManagementEndpoint()));
+        Mono<AccessToken> mono = credential instanceof TenantCredential ?
+                ((TenantCredential) credential).getAccessToken(null, tokenRequestContext) :
+                credential.getToken(tokenRequestContext);
+
+        return mono.flatMap(accessToken ->
+                Mono.zip(AzureResourceManager.authenticate(request -> Mono.just(accessToken), new AzureProfile(env))
```

Comment:
Please refer https://github.com/microsoft/azure-maven-plugins/blob/feature-auth/azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/AzureAppService.java#L135 to set the user agent with track 2 SDK, we may need to do this in the common library

Reply:
user agent should be in global config, let's add a todo comment here(just keep this PR not far away from auth)